### PR TITLE
Bugfix - Unable to update myChatMessageContainer style in MessageThread (#2257)

### PR DESCRIPTION
**Pull request by jboinembalome**
# What

Fixed the inability to change the style of myChatMessageContainer for the MessageThread component.

- Added failedMyChatMessageContainer to the MessageThreadStyles interface.
- Updated the myChatMessageStyle constant according to failedMyChatMessageContainer.

# Why

The `MessageThread `component has a "styles" prop allowing us to change the style of the following elements:
- myChatMessageContainer
- chatMessageContainer

If we add style for both elements we can see that the style is applied for `chatMessageContainer `and not for `myChatMessageContainer`.

Link of the issue: #2257 

# How Tested

We can check the behaviour in the storybook:
- Go to the storybook
- Go to the "Messages with Customized Message Container" section of the `MessageThread `component.

The `MessageThread `will apply the style of `myChatMessageContainer `(fontStyle and boxShadow) instead of applying the style of `FailedMyChatMessageContainer `(red background).

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/MessageThread.tsx [hunk 1]
```diff
@@ -359,9 +361,9 @@ const memoizeAllMessages = memoizeFnAll(
     switch (message.messageType) {
       case 'chat': {
         const myChatMessageStyle =
-          styles?.myChatMessageContainer || message.status === 'failed'
-            ? FailedMyChatMessageContainer
-            : defaultMyChatMessageContainer;
+          message.status === 'failed'
+            ? styles?.failedMyChatMessageContainer || FailedMyChatMessageContainer
```
Comment (JamesBurnside) [inline packages/react-components/src/components/MessageThread.tsx:365]: In the event `failedMyChatMessageContainer` this should fallback to `myChatMessageContainer` (i.e. we consider failedMyChatMessageContainer to be a more specific override).

```suggestion
            ? styles?.failedMyChatMessageContainer ?? styles?.myChatMessageContainer ?? FailedMyChatMessageContainer
```

File: packages/react-components/src/components/MessageThread.tsx [hunk 2]
```diff
@@ -152,6 +152,8 @@ export interface MessageThreadStyles extends BaseCustomStyles {
   chatItemMessageContainer?: ComponentSlotStyle;
   /** Styles for my chat message container. */
   myChatMessageContainer?: ComponentSlotStyle;
+  /** Styles for my chat message container in case of failure.  */
+  failedMyChatMessageContainer?: ComponentSlotStyle;
```
Comment (JamesBurnside) [inline packages/react-components/src/components/MessageThread.tsx:156]: This changes the public API for this component, can you run `rush build:all-flavors` to update the api.md files

**Review by JamesBurnside — CHANGES_REQUESTED** [review]
Thanks for making this contribution! Left some comments.
One extra step along with updating the API files you will need to make is to run `rush changelog`.
This generates a change file that will be included in our changelog when we release a new version.
This change is a `minor` change as we are adding a new field to the message thread component.
And for a changelog entry your PR title is good to use 👍

File: packages/react-components/src/components/MessageThread.tsx (hunk 1 above)
Comment (jboinembalome) [reply packages/react-components/src/components/MessageThread.tsx]: Updated according to your suggestion 👍

File: packages/react-components/src/components/MessageThread.tsx [hunk 3]
```diff
@@ -359,9 +361,9 @@ const memoizeAllMessages = memoizeFnAll(
     switch (message.messageType) {
       case 'chat': {
         const myChatMessageStyle =
-          styles?.myChatMessageContainer || message.status === 'failed'
-            ? FailedMyChatMessageContainer
-            : defaultMyChatMessageContainer;
+          message.status === 'failed'
+            ? styles?.failedMyChatMessageContainer || FailedMyChatMessageContainer
+            : styles?.myChatMessageContainer || defaultMyChatMessageContainer;
```
[the remark this answers is not in the document]
Comment (jboinembalome) [reply packages/react-components/src/components/MessageThread.tsx]: Updated according to your suggestion 👍

File: packages/react-components/src/components/MessageThread.tsx (hunk 2 above)
Comment (jboinembalome) [reply packages/react-components/src/components/MessageThread.tsx:156]: I ran the `rush build:all-flavors` command and pushed the api.md files

**Comment by jboinembalome** [comment]
Thank you @JamesBurnside for checking the PR.
For information, I did not get any changes after running `rush changelog`.

Below is the result:
image

Could it be because I have synchronized the `Azure:main` branch with my branch?

**Review by JamesBurnside — APPROVED** [review]

**Comment by JamesBurnside** [comment]
> Thank you @JamesBurnside James Burnside FTE for checking the PR. For information, I did not get any changes after running `rush changelog`.

It appears we have a bug in our changefile configuration when accepting contributions from a forked repo - we will look into fixing this asap and I will update this thread once we have a fix in

**Review by anjulgarg — APPROVED** [review]

**Comment by JamesBurnside** [comment]
Thanks for your pateience, the `rush changelog` command has now been fixed so you should be good to run and push the changefile to this PR.
Can you also run `rush build:all-flavors` one more time? I would expect updates to `review/stable/react-components.api.md` and `review/stable/communication-react.api.md` files also.

**Comment by JamesBurnside** [comment]
> 
> @JamesBurnside Thanks for fixing the `rush changelog` command, I ran the command and pushed the changefile. Next, I ran the `rush build:all-flavors` command and unfortunately I got no change.

We're unsure why this is happening for you locally. I tried cloning your fork and building locally and it is updating the stable API file. As we are doing a new npm release very soon I'd like to include this change in the release so we've created a clone of this PR #2449 and included you as coauthor.

**Pull request closed without merging**